src/main/java/org/edumips64/core/Converter.java

```diff
@@ -29,6 +29,9 @@
  * */
 public class Converter {
 
+  private static final int OVERFLOW_32 = -0x80000000;
```

Comment:
This should be
```java
private static final int OVERFLOW_32 = -0x10000000;
```

The number represents -2^32. Checking with Python:

```python
In [1]: hex(-2**32)
Out[1]: '-0x100000000'
```

Reply:
Hello. I am not very familiar with python, but it seems like maybe python's typing system doesn't constrain integers like java's does? That hex value has 9 nibbles, which is too large for java int value (I get "Integer number too large" error when I put this value in the code). And also, it seems like for this case since the msb has to be a 1, then the most significant nibble would have to be 8?

Howard

Reply:
Oh yeah, you are right, sorry about that. -2^32 cannot fit an int! You are right, thanks. My mistake was in trying to translate the old value directly, without checking the type bounds.